app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/login/SignInScreen.kt

```diff
@@ -0,0 +1,158 @@
+import androidx.compose.ui.text.input.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import pl.edu.pjatk.goodwill_ninjas.blooddonor_android.R
+import pl.edu.pjatk.goodwill_ninjas.blooddonor_android.navigation.Screen
+@Composable
+fun SignInScreen() {
+    val navController: NavController = rememberNavController()
+    Column(
+            modifier = Modifier
+                .fillMaxWidth(
+                )
+                .padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(12.dp)
+    ) {
+        Title()
+        val localFocusManager = LocalFocusManager.current
+        val emailState = remember { EmailState() }
+        Email(
+                emailState.text, emailState.error,
+                onEmailChanged = {
+                    emailState.text = it
+                    emailState.validate()
+                },
+                onImeAction = {
+                    localFocusManager.moveFocus(FocusDirection.Down)
+                }
+        )
+        val passwordState = remember {
+            PasswordState()
+        }
+        Password(passwordState.text, passwordState.error) {
+            passwordState.text = it
+            passwordState.validate()
+        }
+        SignInButton(enabled = emailState.isValid() && passwordState.isValid(), navController)
+    }
+}
+@Composable
+fun Title() {
+    Text(text = "Zaloguj się")
+}
+@Composable
+fun Email(email: String, error: String?, onEmailChanged: (String) -> Unit, onImeAction: () -> Unit) {
+    Column {
+        TextField(
+                modifier = Modifier.fillMaxWidth(),
+                value = email,
+                onValueChange = { onEmailChanged(it) },
+                label = { Text(text = "Email/Login") },
+                colors = TextFieldDefaults.textFieldColors(
+                        focusedIndicatorColor = Color.Transparent,
+                        unfocusedIndicatorColor = Color.Transparent
+                ),
+                shape = RoundedCornerShape(8.dp),
+                keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Email,
+                        imeAction = ImeAction.Next
+                ),
+                keyboardActions = KeyboardActions(
+                        onNext = {
+
+                            onImeAction()
+                        }
+                ),
+                isError = error != null
+        )
+        error?.let { ErrorField(it) }
+    }
+}
+@Composable
+fun ErrorField(error: String) {
+    Text(
+            text = error,
+            modifier = Modifier.fillMaxWidth(),
+            style = TextStyle(color = MaterialTheme.colors.error)
+    )
+}
+@Composable
+fun Password(password: String, error: String?, onPasswordChanged: (String) -> Unit) {
```

Comment:
This move to separate component please, preferred `components/login/Password.kt`